repair/repair.cc

```diff
@@ -1920,20 +1922,57 @@ future<> repair_service::do_rebuild_replace_with_repair(locator::token_metadata_
             auto& topology = erm->get_token_metadata().get_topology();
             std::unordered_map<dht::token_range, repair_neighbors> range_sources;
             auto nr_tables = get_nr_tables(db, keyspace_name);
-            rlogger.info("{}: started with keyspace={}, source_dc={}, nr_ranges={}, ignore_nodes={}", op, keyspace_name, source_dc, ranges.size() * nr_tables, ignore_nodes);
+            std::unordered_set<gms::inet_address> dc_nodes;
+            sstring ks_source_dc;
+            if (ignored_in_my_dc < 0) {
```

Comment:
Why do we want to do it only for the first iteration (keyspace), won't it mean that dc_nodes will be empty in other iterations, making the check moot? Did you mean to put dc_nodes outside the loop? Why not put this if outside the loop?

Reply:
You're right.  dc_nodes should be defined outside the loop, right with `ignored_in_my_dc`.
It can still be calculated only once of course since the topology and ignore_nodes don't change.
But we should indeed keep the result for all iterations.

Reply:
done in v2